library/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt

```diff
@@ -19,11 +20,21 @@ abstract class HttpSource : CatalogueSource {
 
     override val hasSearchFilters: Boolean = throw Exception("Stub!")
 
+    /**
+     * Type of UserAgent a source needs
+     */
+    protected open val requiredUserAgentType: UserAgentType = UserAgentType.All
```

Comment:
Maybe remove required prefix?

Reply:
it tells us what the value's purpose is. `supportedUserAgentType`?

Reply:
Yeah that works